### PR TITLE
Handle edge case where preference doesn't load in time

### PR DESCRIPTION
**Pull request by surbhiia**
### Note - Build will fail

Using a future version of `splunk-common-android`. That PR is still in review - https://github.com/signalfx/splunk-common-android/pull/27

### Description

Handle edge case where preference doesn't load in time. 

### Checklist

- [x] My code follows the project's coding standards.
- [x] I have run linters/formatters and fixed any issues.
- [x] There are no merge conflicts.
- [x] I have performed a self-review of my code.
- [x] All new and existing tests pass locally.
- [x] I have added license headers to all files.
- [ ] (If applicable) I have added unit tests for my changes.
- [ ] (If applicable) I have updated the sample app for integration testing.
- [ ] (If applicable) I have updated any relevant documentation.

### Generative AI usage

- [ ] GAI was not used (or, no additional notation is required)
- [ ] Coder created a draft manually that was non-substantively modified by GAI (e.g., refactoring was performed by GAI on manually written code)
- [x] GAI was used to create a draft that was subsequently customized or modified
- [ ] Code was generated entirely by GAI

### How to Test These Changes
- Create an edge case where preference file is too large to be loaded in time and see the SDK handle it gracefully without causing any ANR. 
- Ensure the normal cases where preference loads in time work as expected.

### CODE REVIEW

File: integration/agent/api/src/main/kotlin/com/splunk/rum/integration/agent/api/SplunkRum.kt [hunk 1]
```diff
@@ -334,6 +334,14 @@ class SplunkRum private constructor(
 
             val storage = AgentStorage.attach(application)
 
+            // UserManager reads preferences during construction. Do not allow that first preferences access to
+            // block the main thread while the asynchronous preferences load is still in progress
+            if (storage !is AgentStorage || !storage.isPreferencesReady) {
+                Logger.d(TAG, "install() - Preferences are still loading; returning a no-op instance")
+                AgentIntegration.notifyInstallSkipped()
+                return noop
```
[the remark this answers is not in the document]
Comment (surbhiia) [reply integration/agent/api/src/main/kotlin/com/splunk/rum/integration/agent/api/SplunkRum.kt:342]: Re-trial is not possible within current design where `SplunkRum` instance needs to be synchronously initialized and returned to the host app for further use.

**Review by tonzhan2 — APPROVED** [review]

**Review by TomasChladekSL — CHANGES_REQUESTED** [review]
Requesting changes to put this on hold, I've replied in the Slack thread with concerns about this approach.
